v2/internal/system/system_linux.go

```diff
@@ -24,26 +44,10 @@ func (i *Info) discover() error {
 			return err
 		}
 		for _, dep := range dependencies {
-			if dep.Name == "npm" {
-				locallyInstalled := checkNPM()
-				if locallyInstalled.Installed {
-					dep.Installed = true
-					dep.Version = locallyInstalled.Version
-				}
-			}
-			if dep.Name == "docker" {
-				locallyInstalled := checkDocker()
-				if locallyInstalled.Installed {
-					dep.Installed = true
-					dep.Version = locallyInstalled.Version
-				}
-			}
-			if dep.Name == "upx" {
-				locallyInstalled := checkUPX()
-				if locallyInstalled.Installed {
-					dep.Installed = true
-					dep.Version = locallyInstalled.Version
-				}
+			checker := checkerFunctions[dep.Name]
+			if checker != nil {
+				checkLocallyInstalled(checker, dep)
+				continue
```

Comment:
is continue really needed here?

also the `if` on line 48 what protects us against?